families/smallbank/protos/smallbank.proto

```diff
@@ -0,0 +1,126 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// -----------------------------------------------------------------------------
+
+syntax = "proto3";
+option go_package = "smallbank_pb2";
+
+message Account {
+    // Customer ID
+    uint32 customer_id = 1;
+
+    // Customer Name
+    string customer_name = 2;
+
+    // Savings Balance (in cents to avoid float)
+    uint32 savings_balance = 3;
+
+    // Checking Balance (in cents to avoid float)
+    uint32 checking_balance = 4;
+}
+
+message SmallbankTransactionPayload {
+    message CreateAccountTransactionData {
+        // The CreateAccountTransaction creates an account
+
+        // Customer ID
+        uint32 customer_id = 1;
+
+        // Customer Name
+        string customer_name = 2;
+
+        // Initial Savings Balance (in cents to avoid float)
+        uint32 initial_savings_balance = 3;
+
+        // Initial Checking Balance (in cents to avoid float)
+        uint32 initial_checking_balance = 4;
+    }
+
+    message DepositCheckingTransactionData {
+        // The DepositCheckingTransction adds an amount to the customer's
+        // checking account.
+
+        // Customer ID
+        uint32 customer_id = 1;
+
+        // Amount
+        uint32 amount = 2;
+    }
+
+    message WriteCheckTransactionData {
+        // The WriteCheckTransaction removes an amount from the customer's
+        // checking account.
+
+        // Customer ID
+        uint32 customer_id = 1;
+
+        // Amount
+        uint32 amount = 2;
+    }
+
+    message TransactSavingsTransactionData {
+        // The TransactSavingsTransaction adds an amount to the customer's
+        // savings account. Amount may be a negative number.
+
+        // Customer ID
+        uint32 customer_id = 1;
+
+        // Amount
+        int32 amount = 2;
+    }
+
+    message SendPaymentTransactionData {
+        // The SendPaymentTransaction transfers an amount from one customer's
+        // checking account to another customer's checking account.
+
+        // Source Customer ID
+        uint32 source_customer_id = 1;
+
+        // Destination Customer ID
+        uint32 dest_customer_id = 2;
+
+        // Amount
+        uint32 amount = 3;
+    }
+
+    message AmalgamateTransactionData {
+        // The AmalgamateTransaction transfers the entire contents of one
+        // customer's savings account into another customer's checking
+        // account.
+
+        // Source Customer ID
+        uint32 source_customer_id = 1;
+
+        // Destination Customer ID
+        uint32 dest_customer_id = 2;
+    }
+
+    enum PayloadType {
+        CREATE_ACCOUNT = 0;
+        DEPOSIT_CHECKING = 1;
+        WRITE_CHECK = 2;
+        TRANSACT_SAVINGS = 3;
+        SEND_PAYMENT = 4;
+        AMALGAMATE = 5;
+    }
+
+    PayloadType payload_type = 1;
```

Comment:
Remind me why we don't use `oneof` for these?  I know the argument has been made, but I haven't internalized it yet.